### DISCUSSION TITLE
Problem with the msh.12.1 field

### DISCUSSION THREAD

**Discussion by Adrialopezgurrea** [nextgenhealthcare/connect#5832] [Q&A] [answered]
Hi everyone,

First of all sorry if this problem has been solved before, is the first time I'm using this.

I'm using a Mirth Connect channel to read values from a database, create a xml message with these values and then convert it to a HL7 v2.3 message (all of this changes in the source) to send it.

The problem comes when it seems that I'm not using the field "VID.1" in the HL7 message. With the mirth connect version that I had before (maybe it's the administrator launcher version) I didn't have this problem. Instead of "VID.1", the field that appears in my message is "MSH12.1".
Anyone know how to solve this?
Error:
![image](https://github.com/nextgenhealthcare/connect/assets/82399947/a8a90dd1-4c64-4416-a814-f8fd1212fce7)

Outbound message template:

![image](https://github.com/nextgenhealthcare/connect/assets/82399947/05871014-c713-4746-a116-332a08e2775d)

Outbound message tree:

![image](https://github.com/nextgenhealthcare/connect/assets/82399947/7d6c73d6-5544-4328-8386-bbd2fe56abea)

Final message generated:

![image](https://github.com/nextgenhealthcare/connect/assets/82399947/cdf77c21-3995-4b46-b7ba-44dba52b1c6b)

Thank you very much!

**Comment by pacmano1** [comment]
I don't fully understand your question.   Perhaps post the full channel. Also be sure to remove PII and PHI.  A database reader always returns XML, so I am not clear what you mean when you say you are building XML unless your database query itself (not in Mirth) returns XML. You also have hl7 2.5 in your outbound message template, but you said you want to create 2.3.   

I think a channel export would help clarify what you are actually asking.

**Reply by Adrialopezgurrea** [reply]
[channelFujiPACss.txt](https://github.com/nextgenhealthcare/connect/files/11882090/channelFujiPACss.txt)
 this is the channel. I think the error is because I have exported this channel to a new mirth and this is causing problems.

**Reply by Adrialopezgurrea** [reply]
and this is the transformer of the source.
[TransformerExported2.txt](https://github.com/nextgenhealthcare/connect/files/11882135/TransformerExported2.txt)

 sorry for my errors on the version of the hl7 I said, I was trying to solve the problem and tried different things. What do you mean about PII and PHI?
 
 The error:
 ER7Serializer error
ERROR MESSAGE: Error converting XML to ER7
ca.uhn.hl7v2.HL7Exception: Couldn't find VID.1 in message beginning: <HL7Message><MSH><MSH.1>|</MSH.1><MSH.2>^~\&amp;</MSH.2><MSH.3><MSH.3.1>HIS</MSH.3.1></MSH.3><MSH.4><MSH.4.1>HIS</MSH.4.1></MSH.4><MSH.5><MSH.5.1>PACS
	at ca.uhn.hl7v2.parser.XMLParser.parseLeaf(XMLParser.java:775)
	at ca.uhn.hl7v2.parser.XMLParser.getVersion(XMLParser.java:734)
	at ca.uhn.hl7v2.parser.Parser.parse(Parser.java:205)
	at com.mirth.connect.plugins.datatypes.hl7v2.ER7Serializer.fromXML(ER7Serializer.java:246)
	at com.mirth.connect.donkey.server.channel.FilterTransformerExecutor.processConnectorMessage(FilterTransformerExecutor.java:122)
	at com.mirth.connect.donkey.server.channel.DestinationConnector.transform(DestinationConnector.java:402)
	at com.mirth.connect.donkey.server.channel.DestinationChain.doCall(DestinationChain.java:114)
	at com.mirth.connect.donkey.server.channel.DestinationChain.call(DestinationChain.java:63)
	at com.mirth.connect.donkey.server.channel.Channel.process(Channel.java:1836)
	at com.mirth.connect.donkey.server.channel.Channel.dispatchRawMessage(Channel.java:1288)
	at com.mirth.connect.donkey.server.channel.SourceConnector.dispatchRawMessage(SourceConnector.java:192)
	at com.mirth.connect.donkey.server.channel.SourceConnector.dispatchRawMessage(SourceConnector.java:170)
	at com.mirth.connect.connectors.jdbc.DatabaseReceiver.processRecord(DatabaseReceiver.java:315)
	at com.mirth.connect.connectors.jdbc.DatabaseReceiver.processResultSet(DatabaseReceiver.java:205)
	at com.mirth.connect.connectors.jdbc.DatabaseReceiver.poll(DatabaseReceiver.java:149)
	at com.mirth.connect.donkey.server.channel.PollConnectorJob.execute(PollConnectorJob.java:49)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:557)

**Reply by Adrialopezgurrea** [reply]
I didn't know a database reader always returns XML. So the transformation is from xml to a HL7 message, I want to return a hl7 message.

**Reply by pacmano1** [reply]
PII and PHI, that means don't post real patient data by accident which people sometimes do when they forget to modify message templates.   "Protected Health Information (PHI) is any health information that includes any of the 18 elements identified by HIPAA. Personally Identifiable Information (PII) is defined as data used in research that is not considered PHI and is therefore not subject to the HIPAA Privacy and security Rules."    While HIPAA is a USA concept, data privacy applies in the EU and lots of other places as well.

Can you post a sample inbound message?  i.e. the raw source you see for a message the shows the problem.   Remember to de-identify the message.  It should be XML.   I suspect one of your assignments Mirth is intrepreting as a segment.

Also, no need to post here and the forums.   Or, if you do, cross reference the posts.

Also, make your assignments in order (you assign PID-2 at the end), should be after step 5 of the transformer.

**Reply by Adrialopezgurrea** [reply]
[XMLANONIMIZADO.txt](https://github.com/nextgenhealthcare/connect/files/11883099/XMLANONIMIZADO.txt)
That's a sample inbound message. I changed the private information for "REDACTED". I don't know if the problem is in the inbound message because the value of the msh.12.1 is given in the outbound message template and is not taken in the SQL query. I'm sorry if I'm doing something wrong with the posts, is the first time I use this platform.

**Reply by pacmano1** [reply]
I do not get an error with that message.  Also in your channel you commented out the "return" on Scripts tab, deploy drop down, you should restore that.

I suspect you are getting a message from your DB that may, when inserted to the HL7 message, results in breaking the XML.

**Reply by pacmano1** [reply]
You should also clear out values in your oubound message template (the fields I mean) because if you do not overwrite them, the original value will persist.   

I would not normally return hl7 characters (like ^) in a result set also.

**Comment by tonygermano** [comment, marked as answer]
Your outbound data type properties for `Destination 1` are set to use the strict parser. That is what is causing it to look for VID.1. Your inbound properties do not have that box checked, so it's generating a message without VID.1.

`ca.uhn.hl7v2.HL7Exception` is another clue, as that is a HAPI error, which only occurs when you use the strict parser.

**Reply by Adrialopezgurrea** [reply]
Yeah that was it. Thank you very much.

**Reply by pacmano1** [reply]
The exported channel shared above had the strict parsing destination disabled.   That is why the messsage did not fail for me.  While I am glad @tonygermano identified your problem, next time post exactly the same channel as the one you are having a problem with including those destiantions that are enabled.

**Reply by Adrialopezgurrea** [reply]
The strict parsing was disabled in all of the outbound properties except for the destination 1. When I enabled these destination and disabled the destination 2, the problem appeared. I understand what you are saying, sorry.

**Reply by pacmano1** [reply]
No worries, important part is you have the fix.

**Reply by tonygermano** [reply]
@pacmano1 it was in the exported channel that I downloaded. That's how I identified it was in Destination 1. If you only downloaded the source transformer, the problem wasn't in the source.

**Reply by pacmano1** [reply]
I am saying that destination was disabled, from the XML above:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/7cb54c43-f1ad-4cf9-952c-b2abc70a1c38)

**Reply by tonygermano** [reply]
My mistake. I was only looking at the xml and did not import the channel. I missed that the destination was not enabled.
